docs/move_clusters.md

```diff
@@ -0,0 +1,59 @@
+# Moving Hive-managed Clusters
+
+## Summary
+
+There is a need to be able to move Hive-managed clusters from one Hive instance to another.
+This may be useful if the user want to replace a Hive instance with a new one or if the user
+wants to implement sharding between Hive instances.
+
+Hive will manage all of the moving for the user. The user will create a CR indicating with
+details about the destination cluster and label selectors to match against the cluster to
+move.
+
+## ClusterRelocator CRD
+
+A ClusterRelocator is cluster-scoped.
+
+This is an example of a ClusterRelocator that may be used to move all of the ClusterDeployments
+with the label `hive-instance: hive2` to the remote cluster accessed via the kubeconfig secret
+at "hive2-remote-kubecofig-secret" in the "hive" namespace.
+
+```yaml
+apiVersion: hive.openshift.io/v1
+kind: ClusterRelocator
+metadata:
+  name: hive2-cluster-relocator
+spec:
+  kubeconfigSecretRef:
+    name: hive2-remote-kubeconfig-secret
+    namespace: hive
+  clusterDeploymentSelector:
+    matchLabels:
+      hive-instance: hive2
+```
+
+## Process
+
+There is a controller that will reconcile ClusterDeployments and watch ClusterDeployments
+and ClustereRelocators. For ClusterDeployments that match with a ClusterRelocator, the
+controller will move the ClusterDeployment to the Hive instance associated with the kubeconfig
+referenced in the ClusterRelocator.
+
+1. Add the `hive.openshift.io/relocating=<relocator-name>` annotation to the ClusterDeployment. This informs other controllers not to take action on the cluster.
```

Comment:
It would be nice if we could make this more generic as this could be generally useful. Is the relocator name actually needed? Could we go with something like hive.openshift.io/disabled or something similar.

Reply:
The name is needed to cover the case where the ClusterRelocator is deleted in the middle of a relocation. We already have generic ways of disabled via the syncset-pause annotation and the preserveOnDelete field.

Reply:
I did not want to re-use those two things because I wanted to preserve them during the move. For example, if the syncset-pause annotation was set prior to the move, then it should remain set after the move.

Reply:
Those are not quite as ubiquitous as I'd like (MachinePools exempted for example), but if the name is useful then that's ok.

Reply:
| Those are not quite as ubiquitous as I'd like (MachinePools exempted for example)

I don't understand that statement. Could you rephrase it?

Reply:
We can disable syncsets and deprovision, but not remotemachineset controller as an example, I was thinking it might be handy to have a way to disable ALL controllers. (not what this change is about so that can be a discussion for another day if we need it and re-using this annotation feels too weird)

Reply:
The syncset-pause annotation **does** disable the remotemachineset controller.

Reply:
Ah I see, bad example then, but my point was a mechanism to disable *all* controllers.

Reply:
The issue with that is the same as using the syncset-pause annotation for relocation. It requires extra mechanisms to determine whether the annotation was present before relocation and therefore needs to be retained after relocation.

Is there a use case for disabling reconciliation of all controllers outside of the relocation use case?